Classes/ios/Scanners/MTBBarcodeScanner.m

```diff
@@ -174,21 +174,28 @@ - (void)dealloc {
 #pragma mark - Scanning
 
 + (BOOL)cameraIsPresent {
-    return [AVCaptureDevice defaultDeviceWithMediaType:AVMediaTypeVideo] != nil;
+    return [self hasCamera:MTBCameraBack] || [self hasCamera:MTBCameraFront];
 }
 
 + (BOOL)hasCamera:(MTBCamera)camera {
-    AVCaptureDevicePosition position = [self devicePositionForCamera:camera];
+    switch (camera) {
+        case MTBCameraBack:
+            return [self isVideoSupportedForCameraDevice:UIImagePickerControllerCameraDeviceRear];
 
-    for (AVCaptureDevice *device in [AVCaptureDevice devicesWithMediaType:AVMediaTypeVideo]) {
-        if (device.position == position) {
-            return YES;
-        }
+        case MTBCameraFront:
+            return [self isVideoSupportedForCameraDevice:UIImagePickerControllerCameraDeviceFront];
     }
 
+    NSAssert(NO, @"Invalid camera type: %lu", (unsigned long)camera);
+
     return NO;
 }
 
++ (BOOL)isVideoSupportedForCameraDevice:(UIImagePickerControllerCameraDevice)device {
+    // returns YES even if camera access is AVAuthorizationStatusDenied or AVAuthorizationStatusRestricted
```

Comment:
should I just mention AVAuthorizationStatusRestricted here as it is the difference to AVCaptureDevice?